docs/howto/jobs.rst

```diff
@@ -157,3 +157,56 @@ Prevent duplicate job execution by assigning a unique key:
   Unique jobs ensure only one job with the same ``unique_key`` is
   queued or running at a time, but any number can be completed or
   failed.
+
+Concurrency
+-----------------------
+
+Control the number of jobs that can run simultaneously using concurrency:
```

Comment:
We should probably reword this a bit to make it clear immediately that this impacts _global_ concurrency. I see you've got the note down below but it's the core reason to use this. Something like:

"Control the number of jobs with the same concurrency key that can run simultaneously across all workers and queues using `with_concurrency()`:"?